File: app/src/main/java/com/example/whatsdown/Message.java
package com.example.whatsdown;

public class Message {

    String id;
    CurrentUser sender;
    String content;
    String created;
    public Message(String id, CurrentUser sender, String content, String created) {
        this.id = id;
        this.sender = sender;
        this.content = content;
        this.created = created;
    }
    public CurrentUser getSender() { return sender; }
    public String getContent() { return content; }
    public String getCreated() { return created; }
}

File: server/services/message.js
// const Message = require('../models/message');
import Message from '../models/message.js'

const createMessage = async (msg, sender) => {
    let message = new Message({
        sender: sender,
        content: msg,
        created: Date.now()
    });
    let saveMessage = await message.save()
    message.id = saveMessage._id;
    await message.save();
    return message._id;
};

const getAllMessage = async (chat) => {
    const allMsg = [];
    for(let msg of chat.messages){
        let singleMsg = await Message.findById(msg).populate('sender').exec();
        const { username, ...newObjectSender } = singleMsg.sender;
        const newObject = {
            ...singleMsg,
            sender: username
        };
        allMsg.push(newObject);
    }
    return allMsg;
};

const getfullMsgById  = async (id) => {
    let singleMsg = await Message.findById(id).populate('sender').exec();
    return singleMsg;
};

const deleteMsgById  = async (id) => {
    const msg = await Message.findById(id);
    if (!msg)
        return null;
    await msg.deleteOne();
    return msg;
};

export {
    createMessage,
    getAllMessage,
    getfullMsgById,
    deleteMsgById
};